src/Optimizely/DecisionService/DecisionService.php

```diff
@@ -148,6 +150,152 @@ public function getVariation(Experiment $experiment, $userId, $attributes = null
+      if( $experiment == new Experiment()){
+        continue;
+      }
+
+      $variation = $this->getVariation($experiment, $userId, $attributes);
+      if($variation instanceof Variation && $variation != new Variation){
+        $this->_logger->log(Logger::INFO,
+          "The user '{$userId}' is bucketed into experiment '{$experiment->getKey()}' of feature '{$feature_flag_key}'.");
+        return array(
+          "experiment"=> $experiment,
+          "variation" => $variation
+        );
+      }
+    }
+
+    $this->_logger->log(Logger::INFO,
+      "The user '{$userId}' is not bucketed into any of the experiments on the feature '{$feature_flag_key}'.");
+
+    return null;
+  }
+
+   /**
+   * Get the variation if the user is bucketed for one of the rollouts on this feature flag
+   * @param  FeatureFlag $featureFlag The feature flag the user wants to access
+   * @param  string      $userId      user id
+   * @param  array       $attributes  user attributes
+   * @return Variation/null
+   */
+  public function  getVariationForFeatureRollout(FeatureFlag $featureFlag, $userId, $attributes){
+    $feature_flag_key = $featureFlag->getKey();
+    $rollout_id = $featureFlag->getRolloutId();
+    if(empty($rollout_id)){
+       $this->_logger->log(Logger::DEBUG,
+        "Feature flag '{$feature_flag_key}' is not part of a rollout.");
+       return null; 
+    }
+    $rollout = $this->_projectConfig->getRolloutFromId($rollout_id);
+    if($rollout == new Rollout()){
+      return null;
+    }
+
+    $rules = $rollout->getExperiments();
+    if(sizeof($rules) == 0)
+      return null;
+
+    // Evaluate all rollout rules except for last one
+    for($i=0; $i<sizeof($rules)-1; $i++){
+      $experiment = $rules[$i];
+
+      // Evaluate if user meets the audience condition of this rollout rule
+      if (!Validator::isUserInExperiment($this->_projectConfig, $experiment, $attributes)) {
+        $this->_logger->log(
+          Logger::DEBUG,
+          sprintf('User "%s" does not meet conditions to be in experiment "%s".', $userId, $experiment->getKey())
+        );
+        // Evaluate this user for the next rule
+        continue;
+      }
+
+      $this->_logger->log(Logger::DEBUG,
+        sprintf("User '{$userId}' meets conditions for targeting rule '%s'.", $i+1));
+
+      // Evaluate if user satisfies the traffic allocation for this rollout rule
+      $variation = $this->_bucketer->bucket($this->_projectConfig, $experiment, $userId, $userId);
+      if($variation != new Variation()){
+        return $variation;
+      } else {
+        $this->_logger->log(Logger::DEBUG,
+         "User '{$userId}' is not in the traffic group for the targeting rule. Checking 'Eveyrone Else' rule now.");
+        break;
+      }
+    }
+
+    // Evaluate Everyone Else Rule / Last Rule now
+    $experiment = $rules[sizeof($rules)-1];
+    $variation = $this->_bucketer->bucket($this->_projectConfig, $experiment, $userId, $userId);
```

Comment:
we have to ignore audience, are we doing it when passing arguments to bucketer.bucket?

Reply:
Yes. Bucketer does not check audience. In all rules before the last rule, we check for audience first, then call bucketer. In this one, we directly call the bucketer